iceoryx_utils/include/iceoryx_utils/internal/lockfree_queue/unique.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <utility>
+
+namespace iox
+{
+template <typename T>
+class unique
+{
+  public:
+    struct invalid_t
+    {
+        explicit constexpr invalid_t()
+        {
+        }
+    };
+
+    static constexpr invalid_t invalid{};
+
+    template <typename... Args>
+    unique(Args&&... args)
+        : m_value(std::forward<Args>(args)...)
+    {
+    }
+
+    unique()
+        : m_value()
+    {
+    }
+
+    unique(invalid_t)
+        : m_valid(false)
+    {
+    }
+
+    // no copies
+    unique(const unique&) = delete;
+    unique& operator=(const unique&) = delete;
+
+    // only move
+    unique(unique&& other)
+        : m_value(std::move(other.m_value))
+        , m_valid(other.m_valid)
+    {
+        other.m_valid = false;
+    };
+
+    unique& operator=(unique&& other)
+    {
+        if (this != &other)
+        {
+            m_value = std::move(other.m_value);
+            m_valid = other.m_valid;
+            other.m_valid = false;
+        }
+        return *this;
+    }
+
+    const T& operator*() const
+    {
+        return m_value;
+    }
+
+    T release()
+    {
+        m_valid = false;
```

Comment:
What happens when it is in an invalid state? Then this move would be undefined behavior and this is forbidden in our world.